File: micronaut-camunda-bpm-feature/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/RuntimeErrorDelegate.java
package info.novatec.micronaut.camunda.bpm.feature.test;

import org.camunda.bpm.engine.delegate.DelegateExecution;
import org.camunda.bpm.engine.delegate.JavaDelegate;

import javax.inject.Singleton;

/**
 * {@link JavaDelegate} which throws a {@link RuntimeException}.
 *
 * @author Tobias Schäfer
 */
@Singleton
public class RuntimeErrorDelegate implements JavaDelegate {

    @Override
    public void execute(DelegateExecution execution) {
        throw new RuntimeException("RuntimeException thrown by RuntimeErrorDelegate");
    }
}

File: micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/RuntimeErrorDelegate.kt
package info.novatec.micronaut.camunda.bpm.feature.test

import org.camunda.bpm.engine.delegate.DelegateExecution
import org.camunda.bpm.engine.delegate.JavaDelegate
import javax.inject.Singleton

/**
 * [JavaDelegate] which throws a [RuntimeException].
 *
 * @author Tobias Schäfer
 */
@Singleton
class RuntimeErrorDelegate : JavaDelegate {
    override fun execute(execution: DelegateExecution) {
        throw RuntimeException("RuntimeException thrown by RuntimeErrorDelegate")
    }
}